src/io/bin.cpp

```diff
@@ -510,20 +510,24 @@ namespace LightGBM {
 
     if (!is_trivial_) {
       default_bin_ = ValueToBin(0);
+      most_freq_bin_ =
+          static_cast<uint32_t>(ArrayArgs<int>::ArgMax(cnt_in_bin));
       if (bin_type_ == BinType::CategoricalBin) {
-        CHECK(default_bin_ > 0);
+        if (most_freq_bin_ == 0) {
+          CHECK(num_bin_ > 1);
+          // FIXME: how to enable `most_freq_bin_ = 0` for categorical features
```

Comment:
@guolinke Create issue for FIXME to not lose it?

Reply:
^ I agree, we should be using issues instead of `TODO`, `FIXME`, etc.

Reply:
I think this a not trivial, so a `fixme` is here. 
BTW, PR #2463 will rewrite categorical feature solutions, and these codes are not needed.